### PR TITLE
Add RFC-1001 Update user profile

### PR DESCRIPTION
**Pull request by touilleMan**
# What has changed ?

- [ ] Does this PR affect the User ?

### CODE REVIEW

**Review by FirelightFlagboy — CHANGES_REQUESTED** [review]
Holding to give time to merge the previous rfcs and have the correct id :p

**Review by FirelightFlagboy — CHANGES_REQUESTED** [review]

File: docs/rfcs/0038-update-user-profile.md [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+system if it datetime is earlier than the one in the user created's certificate.
+
+We have already encounter this kind of issues with the realm role update system (the famous
+"demi-ballpark" problem ^^), so similar countermeasure should be taken (i.e. ensure the
+inserted certificate's datetime is not in the past compared to other items already in the
+database that may interact with it).
+
+## Client side
+
+Current remote loader system use a 1h cache system. This works ok enough considering
```
[the remark this answers is not in the document]
Comment (touilleMan) [reply docs/rfcs/0038-update-user-profile.md:198]: no this is another cache system ;-)

File: docs/rfcs/0038-update-user-profile.md [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+- The user has profile X, the cache is up to date
+- False positive: The user has profile Y, the cache thinks it has profile X.
+  e.g. The remote loader consider valid a `user_create` that have done after profile has switched from ADMIN to REGULAR.
+- False negative: The user has profile Y, the cache thinks it has profile X.
+  e.g. The remote loader consider invalid a `user_create` that have done after profile has switched from REGULAR to ADMIN.
+
+So on top of false positive, we now also have to handle false negative :(
+
+The solution for this seems to add a special handle in the code when a validation failure
+occurs to send a query to the server in order to make sure we aren't short of a certificate.
```
[the remark this answers is not in the document]
Comment (touilleMan) [reply docs/rfcs/0038-update-user-profile.md:220]: The server takes a bit of time to propagate the new certificates to the clients, so in the meantime a client can try to read a data that appear to him as invalid

File: docs/rfcs/0038-update-user-profile.md [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+While not very complex in it API changes, this RFC has a major impact: it changes the
+postulate that a user's only mutability is it revocation !
+
+This will break most part of the code in unexpected ways if care is not taken.
+
+### Server side
+
+Typical issues we can run into:
+
+1) Server receive an authenticated query for a `user_create` command
```
[the remark this answers is not in the document]
Comment (touilleMan) [reply docs/rfcs/0038-update-user-profile.md:181]: the other items are in the present tense, so no need for past here ;-)

File: docs/rfcs/0038-update-user-profile.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (FirelightFlagboy) [reply docs/rfcs/0038-update-user-profile.md:220]: ok

**Review by FirelightFlagboy — APPROVED** [review]

**Pull request merged**